### PR TITLE
Tournament new and edit

### PR DESCRIPTION
**Pull request by bastrapp**

### CODE REVIEW

File: backend/src/main/java/de/ultical/backend/api/EventsResource.java [hunk 1]
```diff
@@ -91,40 +76,137 @@ public Event getEvent(@PathParam("eventId") int eventId) {
...
+        // first delete the old mapping
+        this.dataStore.removeAllAdminsFromEvent(event);
+
+        for (User admin : event.getAdmins()) {
+            try {
+                this.dataStore.addAdminToEvent(event, admin);
+            } catch (PersistenceException e) {
+                LOG.error("Error adding Admin:\nTeam: {} ( {} )\nUser: {} ( {} )\n",
+                        event.getName(), event.getId(), admin.getFullName(), admin.getId());
+                LOG.error("exception:", e);
```
Comment (bslbckr) [inline backend/src/main/java/de/ultical/backend/api/EventsResource.java]:
```suggestion
                
```

File: backend/src/main/java/de/ultical/backend/api/EventsResource.java [hunk 2]
```diff
@@ -91,40 +76,137 @@ public Event getEvent(@PathParam("eventId") int eventId) {
...
+        // create the admin mapping
+        // first delete the old mapping
+        this.dataStore.removeAllAdminsFromEvent(event);
+
+        for (User admin : event.getAdmins()) {
+            try {
+                this.dataStore.addAdminToEvent(event, admin);
+            } catch (PersistenceException e) {
+                LOG.error("Error adding Admin:\nTeam: {} ( {} )\nUser: {} ( {} )\n",
+                        event.getName(), event.getId(), admin.getFullName(), admin.getId());
```
Comment (bslbckr) [inline backend/src/main/java/de/ultical/backend/api/EventsResource.java]:
```suggestion
                        event.getName(), event.getId(), admin.getFullName(), admin.getId(), e);
```

File: backend/src/main/java/de/ultical/backend/api/EventsResource.java [hunk 3]
```diff
@@ -91,40 +76,137 @@ public Event getEvent(@PathParam("eventId") int eventId) {
...
+            } else if (obj instanceof List) {
+                empty = ((List) obj).isEmpty();
+            }
+        } else {
+            empty = true;
+        }
+        if (empty) {
+            throw new WebApplicationException("Missing event parameters: " + name, Status.EXPECTATION_FAILED);
+        }
+    }
```
Comment (bslbckr) [inline backend/src/main/java/de/ultical/backend/api/EventsResource.java:215]: I would replace this unspecific method by three more specific implementations:
```
private void assureNotEmpty(string obj, string name) { if (obj == null || obj.isEmpty()) throw new WebApplicationException(); }
private void assureNotEmpty(Location obj, string name) { if (obj == null || assureNotEmpty(obj.getCity())) {...}}
private void assureNotEmpty(List obj, string name) { if (obj == null || obj.isEmpty()) {...}}
```

File: backend/src/main/java/de/ultical/backend/data/DataStore.java [hunk 4]
```diff
@@ -911,12 +946,37 @@ public void removeAdminFromTeam(Team team, User admin) {
...
+    }
+
+    private void modifyEventAdmin(Event event, User admin, BiConsumer<Event, User> dbAction) {
+        Objects.requireNonNull(event);
+        Objects.requireNonNull(admin);
+        try {
+            dbAction.accept(event, admin);
+            this.sqlSession.commit();
+        } catch (PersistenceException pe) {
+            this.sqlSession.rollback();
```
Comment (bslbckr) [inline backend/src/main/java/de/ultical/backend/data/DataStore.java:972]:
```suggestion
            logger.error(pe.getCause().getMessage);
            this.sqlSession.rollback();
```

**Review by bslbckr — APPROVED** [review]
don't we need any changes to the database model?

File: backend/src/main/java/de/ultical/backend/api/EventsResource.java (hunk 3 above)
Comment (bastrapp) [reply backend/src/main/java/de/ultical/backend/api/EventsResource.java:215]: mach ich

**Pull request merged**